twoliter/src/cmd/exec.rs

```diff
@@ -0,0 +1,107 @@
+use crate::common::exec;
+use crate::project;
+use crate::tools::install_tools;
+use anyhow::Result;
+use clap::Parser;
+use log::trace;
+use std::path::PathBuf;
+use tokio::process::Command;
+
+/// Run a cargo make command in Twoliter's build environment. Known Makefile.toml environment
+/// variables will be passed-through to the cargo make invocation.
+#[derive(Debug, Parser)]
+pub(crate) struct Exec {
+    /// Path to the project file. Will search for Twoliter.toml when absent.
+    #[clap(long)]
+    project_path: Option<PathBuf>,
+
+    /// Twoliter does not read this from the CARGO_HOME environment variable to avoid any possible
+    /// confusion between a CARGO_HOME set on the system, and the path intended for the Bottlerocket
+    /// build.
+    #[clap(long)]
+    cargo_home: PathBuf,
+
+    /// Cargo make task. E.g. the word "build" if we want to execute `cargo make build`.
+    makefile_task: String,
+
+    /// Uninspected arguments to be passed to cargo make after the target name. For example, --foo
+    /// in the following command : cargo make test --foo.
+    additional_args: Vec<String>,
+}
+
+impl Exec {
+    pub(super) async fn run(&self) -> Result<()> {
+        let project = project::load_or_find_project(self.project_path.clone()).await?;
+        install_tools(project.tools_dir(), false).await?;
+        let makefile_path = project.tools_dir().join("Makefile.toml");
+
+        let mut args = vec![
+            "make".to_string(),
+            "--disable-check-for-updates".to_string(),
+            "--makefile".to_string(),
+            makefile_path.display().to_string(),
+            "--cwd".to_string(),
+            project.project_dir().display().to_string(),
+        ];
+
+        for (key, val) in std::env::vars() {
+            if is_build_system_env(key.as_str()) {
+                trace!("Passing env var {} to cargo make", key);
+                args.push("-e".to_string());
+                args.push(format!("{}={}", key, val));
+            }
+        }
+
+        args.push("-e".to_string());
+        args.push(format!("CARGO_HOME={}", self.cargo_home.display()));
+        args.push(self.makefile_task.clone());
+
+        for cargo_make_arg in &self.additional_args {
+            args.push(cargo_make_arg.clone());
+        }
+
+        exec(Command::new("cargo").args(args)).await
+    }
+}
+
+/// A list of environment variables that don't conform to naming conventions but need to be passed
+/// through to the `cargo make` invocation.
+const ENV_VARS: [&str; 13] = [
+    "ALLOW_MISSING_KEY",
+    "AMI_DATA_FILE_SUFFIX",
+    "BOOT_CONFIG",
+    "BOOT_CONFIG_INPUT",
+    "CARGO_MAKE_CARGO_ARGS",
+    "CARGO_MAKE_DEFAULT_TESTSYS_KUBECONFIG_PATH",
+    "CARGO_MAKE_TESTSYS_ARGS",
+    "CARGO_MAKE_TESTSYS_KUBECONFIG_ARG",
```

Comment:
Should we add a filter like this below?
```
key.starts_with("CARGO_MAKE")
```

Reply:
No, I don't think so. There are a lot of `CARGO_MAKE_*` variables that `cargo make` sets in the environment, possibly for it's own purposes, and I don't think we want to mess with those. I'm not exactly sure if it would behave differently in the presence of the user setting those. In fact I wonder if I should delete `CARGO_MAKE_CARGO_ARGS` but it seems to be working fine.